src/CodeGen_Hexagon.cpp

```diff
@@ -542,6 +543,18 @@ void CodeGen_Hexagon::init_module() {
         // runtime module that uses popcounth, and horizontally add
         // each pair of lanes.
     };
+
+    if (is_v62) {
```

Comment:
I am leaning towards a flag/predicate in HexagonOptimize.cpp that apply_patterns should check before "applying" a pattern. This way the Pattern Constructor can default to v60 and for all new architectures that are backwards compatible, we can have a flag like so
enum ArchOps {
    v60 = 0x1, 
    v62 = 0x1 << 1;
};
struct Pattern {
   int arch_available_on;
  Pattern(const string &intrin, Expr p, int flags = 0, int arch_available_on = Pattern::v60)
        : intrin(intrin), pattern(p), flags(flags), arch_available_on(arch_available_on) {}

 }
Pattern p{"halide.hexagon...", wild_u16x+..., Pattern::Reinterleaveop0, Pattern::v60 | Pattern::v62};